src/Components/PaginationNav/PaginationNav.tsx

```diff
@@ -0,0 +1,205 @@
+// Libraries
+import React, {FunctionComponent, useState, useEffect, useRef} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {PaginationItem} from './PaginationItem'
+import {PaginationDirectionItem} from './PaginationDirectionItem'
+import {PaginationTruncationItem} from './paginationTruncationItem'
+
+// Styles
+import './Pagination.scss'
+
+// Types
+import {StandardFunctionProps, DirectionType} from '../../Types'
+
+export interface PaginationNavProps extends StandardFunctionProps {
+  /** Total nuber of pages there exists */
+  totalPages: number
+  /** currently active page */
+  currentPage?: number
+  /** Function to be called on page change */
+  onChange: (page: number) => void
+  /** Determines how many pages are displayed within the nav.
+   * 1 = 7 {1,...,4,5,6,...20},
+   * 2 = 9 {1,...,4,5,6,7,8...20},
+   * 3 = 11 {1,...,4,5,6,7,8,9,10...20} and so on */
```

Comment:
The description here is a bit confusing.
```suggestion
  /** Number of page numbers to display to the left/right of the current page number. */
```